src/meshes/t8code_mesh.jl

```diff
@@ -446,35 +428,234 @@ function T8codeMesh(conn::Ptr{p8est_connectivity}; kwargs...)
+                                                                                                             initial_refinement_level,
+                                                                                                             NDIMS,
+                                                                                                             RealT,
+                                                                                                             boundary_symbols)
+    end
+
+    if typeof(connectivity) <: Ptr{p4est_connectivity}
+        cmesh = t8_cmesh_new_from_p4est(connectivity, mpi_comm(), 0)
+    elseif typeof(connectivity) <: Ptr{p8est_connectivity}
+        cmesh = t8_cmesh_new_from_p8est(connectivity, mpi_comm(), 0)
+    else
+        throw("`connectivity` is not of type `Ptr{p*est_connectivity}`.")
+    end
+
+    do_face_ghost = mpi_isparallel()
+    scheme = t8_scheme_new_default_cxx()
+    forest = t8_forest_new_uniform(cmesh, scheme, initial_refinement_level, do_face_ghost,
+                                   mpi_comm())
+
+    return T8codeMesh{NDIMS}(forest, tree_node_coordinates, nodes,
+                             boundary_names, "")
+end
+
+"""
+T8codeMeshCubedSphere(trees_per_face_dimension, layers, inner_radius, thickness;
+                      polydeg, RealT=Float64, initial_refinement_level=0)
+
+Construct a cubed spherical shell of given inner radius and thickness as `T8codeMesh` with
+`6 * trees_per_face_dimension^2 * layers` trees. The mesh will have two boundaries,
+`:inside` and `:outside`.
+
+# Arguments
+- `lat_lon_levels_per_face_dimension::Integer`: number of trees per patch in longitudinal
+                                                and latitudinal direction given as level of
+                                                refinement.
+- `layers::Integer`: the number of trees in the third local dimension of each face, i.e.,
+                     the number of layers of the shell.
+- `inner_radius::Float64`: Radius of the inner side of the shell.
+- `thickness::Float64`: Thickness of the shell. The outer radius will be
+                        `inner_radius + thickness`.
+- `polydeg::Integer`: polynomial degree used to store the geometry of the mesh.
+                      The mapping will be approximated by an interpolation polynomial
+                      of the specified degree for each tree.
+- `RealT::Type`: the type that should be used for coordinates.
+- `initial_refinement_level::Integer`: refine the mesh uniformly to this level before the
+                                       simulation starts.
+"""
+function T8codeMeshCubedSphere(lat_lon_levels_per_face_dimension, layers, inner_radius,
+                               thickness;
+                               polydeg, RealT = Float64, initial_refinement_level = 0)
+    NDIMS = 3
+    cmesh = t8_cmesh_new_cubed_spherical_shell(inner_radius, thickness,
```

Comment:
> Update: This PR does not actually need the latest t8code v2.0.0.

What about this function? Looks like this is introduced in t8code v2.0.0, cf. https://github.com/DLR-AMR/T8code.jl/pull/64/files#diff-692f951b75c5653e27e7079840135fd829a8c6d9fd726791428db72d63573817R12605, right?

Reply:
Ups, this completely slipped out of my mind!

Reply:
I removed the cubed spherical shell code which was included for @benegee elixirs. This code will be part of the PR where the t8code v2.0.0. is introduced to Trixi.

Reply:
Alright, thanks!